src/test/regress/sql/multi_metadata_sync.sql

```diff
@@ -573,8 +575,10 @@ DROP TABLE mx_table;
 \c - postgres - :master_port
 INSERT INTO pg_dist_placement SELECT * FROM pg_dist_placement_temp;
 INSERT INTO pg_dist_partition SELECT * FROM pg_dist_partition_temp;
+INSERT INTO citus.pg_dist_object SELECT * FROM pg_dist_object_temp ON CONFLICT ON CONSTRAINT pg_dist_object_pkey DO NOTHING;
```

Comment:
why do we need `on coflict` here? Isn't `pg_dist_object` already empty?

Reply:
We can have some while creating the distributed table

Reply:
Aren't we dropping all the objects with `MetadataDropCommands` already?

Reply:
Yes we do but creating a distributed table and syncing metadata adds them back

Reply:
Hmm, I'm still not sure. Would it be fixed with your drop sequence improvements? Can you share which objects are conflicting if we don't have ON CONFLICT?

(I'm suspecting something might be overlooked, because why would we insert the same object twice? Or, as the sequence bug fix, maybe there is some other object that is not removed etc.)

Reply:
this is public schema, and as we don't drop schemas from the workers